src/mongocrypt-opts.c

```diff
@@ -48,6 +48,13 @@ _mongocrypt_opts_kms_provider_gcp_cleanup (
    _mongocrypt_buffer_cleanup (&kms_provider_gcp->private_key);
 }
 
+static void
+_mongocrypt_opts_kms_provider_kmip_cleanup (
```

Comment:
The other functions are *_destroy(), but we are *_cleanup()? Is this to distinguish releasing the entire structure from part of it? If so, why not call the *_destroy() function at the call site? Are these expected to grow additional fields? (It might be good to make a note to that effect.) I don't mind this, just seeking clarity.

Reply:
> The other functions are *_destroy(), but we are *_cleanup()? Is this to distinguish releasing the entire structure from part of it?

Precisely.

> Are these expected to grow additional fields?

I think there is a possibility that fields get added in the future. But that has not happened on the AWS, Azure, GCP, or local KMS providers.

> If so, why not call the *_destroy() function at the call site?

True, there is no strong reason for a separate function. Updated to remove `_mongocrypt_opts_kms_provider_kmip_cleanup` and call `_mongocrypt_endpoint_destroy` directly in `_mongocrypt_opts_cleanup`.